### PR TITLE
Update validation docs

### LINKED ISSUE #351: Update Validation section in README
- Should explain usage of `cffconvert`.

### PR DESCRIPTION
**Pull request by sdruskat**
**Related issues**

Fix #351

(For autoclosure of issues when PR is merged use `Fixes #<issue-number>` syntax)

**Describe the changes made in this pull request**

- Replace validation documentation to use `validator.py` with usage of `cffconvert` with Python and Docker.
- Update tools list (add Docker line and `cffconvert docker` links, remove `validator.py`)

**Review checklist**

- [x] Please check if the pull request is against the correct branch  
(format/schema/semantic documentation changes: `develop`; typos, meta files, etc.: `main`)
- [x] Please check if all changes are recorded in `CHANGELOG.md` and adapt if necessary.
- [x] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout 351-update-validation-docs
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
<do other things>
```

- [x] Read through the changes carefully.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,28 +56,26 @@ When you do this, great things may happen:
...
 ## Validation :heavy_check_mark:
 
-You can validate your `CITATION.cff` file with the [Python script](examples/validator.py)
-that we also use for testing the Citation File Format schema itself, like so:
+You can validate your `CITATION.cff` file on the command line with the [`cffconvert` Python package](https://pypi.org/project/cffconvert/):
 
 ```shell
-# clone this repository
-git clone https://github.com/citation-file-format/citation-file-format.git
+# Install cffconvert with pip in userspace
```
Comment (jspaaks) [inline README.md:62]:
```suggestion
# Install cffconvert with pip in user space
```

File: README.md [hunk 2]
```diff
@@ -56,28 +56,26 @@ When you do this, great things may happen:
...
-# change into the cloned repository directory
-cd citation-file-format
-
-# install the validation dependencies in user space
-python3 -m pip install --user ruamel.yaml jsonschema
-
-# run the validator on your CITATION.cff
-python3 examples/validator.py -s schema.json -d path/to/your/CITATION.cff
+# Validate your CFF file
+cffconvert --validate -i <path-to-your-CITATION.cff>
```
Comment (jspaaks) [inline README.md:66]: perhaps we should leave out the `-i`, similar to how the dockerized version is explained

Comment (sdruskat) [reply README.md:66]: Ah, I see, you don't even need the name of the file. Missed that, sorry.

```suggestion
cffconvert --validate
```

**Comment by sdruskat** [comment]
Thanks for the review @jspaaks! I'll merge once the tests pass.

**Pull request merged**